### ISSUE TITLE
Support #exists?

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/query_relation#2]

**Comment by kbrock** [comment]
were you thinking `exists` would be supported by `query_relation` or the implementer of `search`?

**Comment by Fryguy** [comment]
I was thinking query_relation and also in the the querable mixin.

**Comment by Fryguy** [comment]
Thinking about it more though, I never considered search, but it could be a new mode.  Would allow the search writer to write a more efficient exists method if they want it.  For example, against a REST API, maybe they want to send a HEAD request.

**Comment by kbrock** [comment]
The implementer implements `first` and `last`. Their implementation is much more optimized than `all.first`. But maybe `exist?` isn't that different from `first`

Don't think there is a simple way for the implementer to register which methods are available from this library. Since this is already so magical, the bran has very little room left for learning the programming interface.

**Comment by Fryguy** [comment]
Yeah I was originally thinking that `exists?` could be naively implemented as `!!first` or `limit(1).any?` or something.  But that precludes the search writer from doing anything more efficient.

**Comment by kbrock** [comment]
so maybe this is `limit(1).pluck(:id)` ?

**Comment by Fryguy** [comment]
Query relations objects don't necessarily have ids or even attributes for that matter (see the example in the README that's just integers.

**Comment by kbrock** [comment]
ok so `first.exist?`

**Comment by Fryguy** [comment]
yeah read back - it could be `first.present?`, or `limit(1).present?`, or perhaps we let the `:exists` be a new mode that's sent into search and the caller can implement how they want.
